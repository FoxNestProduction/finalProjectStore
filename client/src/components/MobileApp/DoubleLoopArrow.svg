<svg xmlns="http://www.w3.org/2000/svg" width="179" height="165" viewBox="0 0 179 165" fill="none">
    <path d="M174.632 61.1592C174.861 60.4724 174.476 59.7315 173.772 59.5044C173.069 59.2773 172.312 59.6499 172.083 60.3367L174.632 61.1592ZM139.518 93.1575L139.732 94.4514L139.518 93.1575ZM99.285 71.2522L100.625 71.248L99.285 71.2522ZM111.336 57.0156L111.435 58.3222L111.336 57.0156ZM124.128 67.4023L122.868 67.8425L124.128 67.4023ZM89.9995 102.591L90.1526 103.893L89.9995 102.591ZM53.3537 80.017L54.6915 79.9482L53.3537 80.017ZM62.071 69.0496L62.4066 70.3192L62.071 69.0496ZM35.6717 114.287L35.7717 112.981L35.6717 114.287ZM3.45371 94.7338L3.75154 93.4582C3.27576 93.35 2.77865 93.5028 2.4538 93.8572C2.12893 94.2116 2.02769 94.7115 2.18947 95.1622L3.45371 94.7338ZM15.3454 98.782C16.0669 98.9462 16.7852 98.5081 16.9497 97.8036C17.1142 97.099 16.6626 96.3949 15.941 96.2308L15.3454 98.782ZM6.53677 107.274C6.78214 107.957 7.54708 108.32 8.2453 108.083C8.94351 107.846 9.31061 107.1 9.06524 106.417L6.53677 107.274ZM173.358 60.748C172.083 60.3367 172.083 60.3364 172.084 60.3362C172.084 60.3363 172.084 60.3362 172.084 60.3363C172.083 60.3366 172.083 60.3374 172.083 60.3385C172.082 60.3408 172.081 60.3449 172.079 60.3506C172.075 60.362 172.069 60.3802 172.06 60.405C172.043 60.4546 172.016 60.5304 171.98 60.6308C171.908 60.8316 171.797 61.1308 171.647 61.5152C171.347 62.2844 170.889 63.3939 170.264 64.7399C169.012 67.4344 167.096 71.0625 164.441 74.7998C159.112 82.3003 150.927 90.0732 139.303 91.8637L139.732 94.4514C152.388 92.5021 161.124 84.0703 166.645 76.3009C169.414 72.4033 171.406 68.6302 172.705 65.8322C173.356 64.4319 173.834 63.2724 174.152 62.4591C174.311 62.0523 174.429 61.7319 174.509 61.5109C174.548 61.4004 174.578 61.3148 174.599 61.2556C174.609 61.226 174.617 61.2031 174.623 61.1869C174.625 61.1789 174.627 61.1725 174.629 61.1679C174.63 61.1655 174.63 61.1637 174.631 61.1622C174.631 61.1615 174.631 61.1607 174.632 61.1604C174.632 61.1597 174.632 61.1592 173.358 60.748ZM139.303 91.8637C130.227 93.2615 120.492 91.628 113.049 87.8247C105.587 84.012 100.681 78.1764 100.625 71.248L97.945 71.2564C98.0123 79.5266 103.869 86.0843 111.816 90.145C119.781 94.215 130.101 95.9348 139.732 94.4514L139.303 91.8637ZM100.625 71.248C100.608 69.2159 101.47 66.0486 103.325 63.334C105.163 60.6444 107.855 58.5641 111.435 58.3222L111.237 55.7089C106.586 56.0231 103.233 58.7417 101.094 61.8717C98.9723 64.9767 97.9238 68.6497 97.945 71.2564L100.625 71.248ZM111.435 58.3222C114.818 58.0937 117.239 59.3985 119.047 61.2831C120.895 63.2102 122.091 65.7397 122.868 67.8425L125.388 66.9621C124.564 64.7342 123.217 61.8099 120.997 59.4961C118.737 57.1398 115.562 55.4167 111.237 55.7089L111.435 58.3222ZM122.868 67.8425C125.97 76.2277 122.663 83.9741 115.992 90.0208C109.299 96.0866 99.3708 100.263 89.8464 101.289L90.1526 103.893C100.187 102.812 110.665 98.4291 117.819 91.9455C124.993 85.4428 128.972 76.6489 125.388 66.9621L122.868 67.8425ZM89.8464 101.289C81.6769 102.17 73.0456 100.684 66.3844 97.0282C59.7511 93.3875 55.1432 87.6505 54.6915 79.9482L52.0158 80.0857C52.5315 88.8789 57.8374 95.3353 65.0806 99.3109C72.296 103.271 81.5039 104.825 90.1526 103.893L89.8464 101.289ZM54.6915 79.9482C54.5498 77.5315 55.1447 75.4855 56.3831 73.8854C57.6239 72.2821 59.5963 71.0167 62.4066 70.3192L61.7354 67.78C58.4613 68.5926 55.9181 70.1342 54.2429 72.2987C52.5652 74.4664 51.8436 77.149 52.0158 80.0857L54.6915 79.9482ZM62.4066 70.3192C64.9794 69.6807 67.0592 70.1916 68.813 71.4239C70.6217 72.6948 72.1508 74.7859 73.3973 77.3868L75.8233 76.2849C74.4765 73.4745 72.7007 70.9367 70.3705 69.2994C67.9855 67.6235 65.0922 66.9469 61.7354 67.78L62.4066 70.3192ZM73.3973 77.3868C77.5738 86.1018 73.1561 95.2387 64.8458 102.261C56.5568 109.265 44.8822 113.69 35.7717 112.981L35.5717 115.593C45.5673 116.37 57.9208 111.583 66.604 104.246C75.2657 96.927 80.7643 86.595 75.8233 76.2849L73.3973 77.3868ZM35.7717 112.981C28.7 112.431 20.9182 107.767 14.7636 103.044C11.7184 100.707 9.1319 98.4038 7.30654 96.684C6.39454 95.8247 5.67428 95.1128 5.18363 94.6174C4.93835 94.3697 4.75057 94.1763 4.62505 94.0458C4.56228 93.9806 4.5151 93.9311 4.48408 93.8984C4.46856 93.882 4.4571 93.8699 4.44974 93.8621C4.44607 93.8582 4.44341 93.8554 4.4418 93.8537C4.44101 93.8528 4.44045 93.8522 4.44017 93.8519C4.44003 93.8518 4.44001 93.8517 4.43995 93.8517C4.43999 93.8517 4.4401 93.8518 3.45371 94.7338C2.46731 95.6158 2.46755 95.616 2.46786 95.6164C2.46805 95.6166 2.46843 95.617 2.46881 95.6174C2.46958 95.6182 2.4706 95.6193 2.47189 95.6207C2.47447 95.6234 2.4781 95.6272 2.48275 95.6322C2.49207 95.6421 2.50552 95.6563 2.52305 95.6748C2.55812 95.7117 2.60947 95.7656 2.6765 95.8353C2.81057 95.9747 3.00744 96.1775 3.26236 96.4348C3.77213 96.9495 4.51448 97.6832 5.45135 98.5658C7.32372 100.33 9.97995 102.696 13.1148 105.102C19.3202 109.863 27.6473 114.976 35.5717 115.593L35.7717 112.981ZM3.15588 96.0094L15.3454 98.782L15.941 96.2308L3.75154 93.4582L3.15588 96.0094ZM2.18947 95.1622L6.53677 107.274L9.06524 106.417L4.71794 94.3054L2.18947 95.1622Z" fill="#6C5FBC"/>
</svg>