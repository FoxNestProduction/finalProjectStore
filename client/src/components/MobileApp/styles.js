export const flexCenter = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
};

export const section = {
  display: 'flex',
  alignItems: 'center',
  justifyContent: {
    mobile: 'center',
    lgTablet: 'space-around',
  },
  flexDirection: {
    mobile: 'column',
    lgTablet: 'row-reverse',
  },
  mb: {
    mobile: '66px',
    tablet: '85px',
    desktop: '70px',
  },
};

export const descriptionBox = {
  display: 'flex',
  alignItems: {
    mobile: 'center',
    tablet: 'flex-start',
  },
  justifyContent: 'center',
  flexDirection: 'column',
  rowGap: {
    mobile: '35px',
    desktop: '45px',
  },
};

export const list = {
  pl: '23px',
  listStyle: 'disc',
  color: 'text.primary',
  display: 'flex',
  flexDirection: 'column',
  rowGap: {
    mobile: '16px',
    teblet: '20px',
    desktop: '32px',
  },
};

export const listText = {
  fontSize: {
    mobile: '14px',
    tablet: '16px',
    desktop: '24px',
  },
};

export const googleAppleBox = {
  position: 'relative',
  width: '100%',
  display: 'flex',
  flexDirection: {
    mobile: 'column',
    tablet: 'row',
  },
  mb: '25px',
};

export const googleAppleBtn = {
  width: '120px',
  height: '40px',
};

export const arrow = {
  position: 'absolute',
  transform: {
    mobile: 'rotate(97deg)',
    tablet: 'rotate(-20deg)',
  },
  width: {
    mobile: '100px',
    tablet: '110px',
    desktop: '165px',
  },
  bottom: {
    mobile: '-140px',
    tablet: '-30px',
    desktop: '-30px',
  },
  right: {
    mobile: '60%',
    tablet: '-10%',
    lgtablet: '10%',
    desktop: '15%',
  },
};

export const mobileImg = {
  position: 'relative',
  left: {
    mobile: '15px',
    tablet: '0',
  },
  width: {
    mobile: '170px',
    tablet: '216px',
    desktop: '300px',
  },
  mt: {
    mobile: '110px',
    tablet: '0',
  },
};
