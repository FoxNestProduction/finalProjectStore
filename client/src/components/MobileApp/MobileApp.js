import { Divider, Box, Button, Container, List, ListItem, Typography } from '@mui/material';
import React from 'react';
import { section, descriptionBox, list, listText, googleAppleBtn, googleAppleBox, arrow, mobileImg } from './styles';

const MobileApp = () => {
  const onClick = (url) => {
    window.open(url, '_blank');
  };

  return (
    <>
      <Container
        component="section"
        sx={section}
      >
        <Box sx={descriptionBox}>
          <Typography component="h2" variant="h2" sx={{ color: 'text.primary' }} textTransform="capitalize" fontWeight={500}>
            Premium Quality
            <br />
            For Your Health
          </Typography>
          <List sx={list}>
            <ListItem sx={{ display: 'list-item', padding: '0' }}>
              <Typography component="p" variant="subtitle1" sx={listText}>
                Premium quality food is made with ingredients that
                <br />
                are packed with essential vitamins, minerals.
              </Typography>
            </ListItem>
            <ListItem sx={{ display: 'list-item', padding: '0' }}>
              <Typography component="p" variant="subtitle1" sx={listText}>
                These foods promote overall wellness by
                <br />
                support healthy digestion and boosting immunity
              </Typography>
            </ListItem>
          </List>
          <Box sx={googleAppleBox}>
            <Button sx={{ padding: '0' }} onClick={() => onClick('https://play.google.com/')}>
              <Box
                component="img"
                src={`${process.env.PUBLIC_URL}/img/google-play.png`}
                alt="google play"
                sx={googleAppleBtn}
              />
            </Button>
            <Button sx={{ padding: '0' }} onClick={() => onClick('https://www.apple.com/ua/app-store/')}>
              <Box
                component="img"
                src={`${process.env.PUBLIC_URL}/img/app-store.png`}
                alt="google play"
                sx={googleAppleBtn}
              />
            </Button>
            {/* <Box sx={arrow}><DoubleLoopArrow /></Box> */}
          </Box>
        </Box>
        <Box
          component="img"
          src={`${process.env.PUBLIC_URL}/img/layout/mobile.png`}
          alt="phone"
          sx={mobileImg}
        />
      </Container>
      <Container>
        <Divider sx={{ marginBottom: { desktop: '120px', tablet: '70px', mobile: '80px' } }} />
        {/* Додає сіру лінію під секцією довжина якої регул. котейнером */}
      </Container>
    </>
  );
};

export default MobileApp;
