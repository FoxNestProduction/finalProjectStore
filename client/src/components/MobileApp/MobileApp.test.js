import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import MobileApp from "./MobileApp";

describe('MobileApp component', () => {
  test('should render MobileApp component', () => {
    const { asFragment } = render(
          <MobileApp />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('calls onClick with correct URL when Google Play button is clicked', () => {
    render(<MobileApp />);

    const googlePlayButton = screen.getByAltText('google play');
    const openSpy = jest.spyOn(window, 'open');

    fireEvent.click(googlePlayButton);

    expect(openSpy).toHaveBeenCalledWith('https://play.google.com/', '_blank');

    openSpy.mockRestore();
  })

  test('calls onClick with correct URL when App Store button is clicked', () => {
    render(<MobileApp />);

    const appStoreButton = screen.getByAltText('app store');
    const openSpy = jest.spyOn(window, 'open');

    fireEvent.click(appStoreButton);

    expect(openSpy).toHaveBeenCalledWith('https://www.apple.com/ua/app-store/', '_blank');

    openSpy.mockRestore();
  })
});