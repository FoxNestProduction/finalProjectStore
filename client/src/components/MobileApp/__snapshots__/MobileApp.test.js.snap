// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileApp component should render MobileApp component 1`] = `
<DocumentFragment>
  <section
    class="MuiContainer-root MuiContainer-maxWidthLg css-1osigg8-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-enz7jw"
    >
      <h2
        class="MuiTypography-root MuiTypography-h2 css-1ozfjgs-MuiTypography-root"
      >
        Premium Quality
        <br />
        For Your Health
      </h2>
      <ul
        class="MuiList-root MuiList-padding css-1ejqyj9-MuiList-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-txxhdm-MuiListItem-root"
        >
          <p
            class="MuiTypography-root MuiTypography-subtitle1 css-331lyw-MuiTypography-root"
          >
            Premium quality food is made with ingredients that
            <br />
            are packed with essential vitamins, minerals.
          </p>
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-txxhdm-MuiListItem-root"
        >
          <p
            class="MuiTypography-root MuiTypography-subtitle1 css-331lyw-MuiTypography-root"
          >
            These foods promote overall wellness by
            <br />
            support healthy digestion and boosting immunity
          </p>
        </li>
      </ul>
      <div
        class="MuiBox-root css-1jl7xfy"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-16rr57b-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="google play"
            class="MuiBox-root css-77cvsc"
            src="/img/google-play.png"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-16rr57b-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          <img
            alt="app store"
            class="MuiBox-root css-77cvsc"
            src="/img/app-store.png"
          />
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
    <img
      alt="phone"
      class="MuiBox-root css-1g2205e"
      src="/img/layout/mobile.png"
    />
  </section>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <hr
      class="MuiDivider-root MuiDivider-fullWidth css-5fx7t7-MuiDivider-root"
    />
  </div>
</DocumentFragment>
`;
